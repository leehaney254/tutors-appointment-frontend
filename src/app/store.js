import { configureStore } from '@reduxjs/toolkit';
import registerReducer from '../features/RegisterSlice';
import loginReducer from '../features/LoginSlice';

export default configureStore({
  reducer: {
    register: registerReducer,
    login: loginReducer,
  },
});
