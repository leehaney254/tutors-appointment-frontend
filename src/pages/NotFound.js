import React from 'react';
import { Link } from 'react-router-dom';

const NotFound = () => (
  <div>
    <div className="flex items-center justify-center w-screen h-screen">
      <div className="px-4 lg:py-12">
        <div className="lg:gap-4 lg:flex">
          <div className="flex flex-col items-center justify-center md:py-24 lg:py-32">
            <h1 className="font-bold text-blue-600 text-9xl">404</h1>
            <p className="mb-2 text-2xl font-bold text-center text-gray-800 md:text-3xl">
              <span className="text-red-500">Oops!</span>
              {' '}
              Page not found
            </p>
            <p className="mb-8 text-center text-gray-500 md:text-lg">
              The page you’re looking for doesn’t exist.
            </p>
            <Link
              to="/"
              className="px-6 py-2 text-sm font-semibold text-blue-800 bg-blue-100"
            >
              Go home
            </Link>
          </div>
          <div className="mt-4">
            <img
              src="https://cdn.pixabay.com/photo/2016/11/22/23/13/black-dog-1851106__340.jpg"
              alt="img"
              className="object-cover w-full h-full"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
);

export default NotFound;
