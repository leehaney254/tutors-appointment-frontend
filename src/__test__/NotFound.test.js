import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import store from '../app/store';
import NotFound from '../pages/NotFound';

it('should render AddTutor page', () => {
  const notFound = render(
    <BrowserRouter>
      <Provider store={store}>
        <NotFound />
      </Provider>
    </BrowserRouter>,
  );
  expect(notFound).toMatchSnapshot();
});
