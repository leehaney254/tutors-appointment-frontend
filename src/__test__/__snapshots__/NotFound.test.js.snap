// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render AddTutor page 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <div
          class="flex items-center justify-center w-screen h-screen"
        >
          <div
            class="px-4 lg:py-12"
          >
            <div
              class="lg:gap-4 lg:flex"
            >
              <div
                class="flex flex-col items-center justify-center md:py-24 lg:py-32"
              >
                <h1
                  class="font-bold text-blue-600 text-9xl"
                >
                  404
                </h1>
                <p
                  class="mb-2 text-2xl font-bold text-center text-gray-800 md:text-3xl"
                >
                  <span
                    class="text-red-500"
                  >
                    Oops!
                  </span>
                   
                  Page not found
                </p>
                <p
                  class="mb-8 text-center text-gray-500 md:text-lg"
                >
                  The page you’re looking for doesn’t exist.
                </p>
                <a
                  class="px-6 py-2 text-sm font-semibold text-blue-800 bg-blue-100"
                  href="/"
                >
                  Go home
                </a>
              </div>
              <div
                class="mt-4"
              >
                <img
                  alt="img"
                  class="object-cover w-full h-full"
                  src="https://cdn.pixabay.com/photo/2016/11/22/23/13/black-dog-1851106__340.jpg"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <div
        class="flex items-center justify-center w-screen h-screen"
      >
        <div
          class="px-4 lg:py-12"
        >
          <div
            class="lg:gap-4 lg:flex"
          >
            <div
              class="flex flex-col items-center justify-center md:py-24 lg:py-32"
            >
              <h1
                class="font-bold text-blue-600 text-9xl"
              >
                404
              </h1>
              <p
                class="mb-2 text-2xl font-bold text-center text-gray-800 md:text-3xl"
              >
                <span
                  class="text-red-500"
                >
                  Oops!
                </span>
                 
                Page not found
              </p>
              <p
                class="mb-8 text-center text-gray-500 md:text-lg"
              >
                The page you’re looking for doesn’t exist.
              </p>
              <a
                class="px-6 py-2 text-sm font-semibold text-blue-800 bg-blue-100"
                href="/"
              >
                Go home
              </a>
            </div>
            <div
              class="mt-4"
            >
              <img
                alt="img"
                class="object-cover w-full h-full"
                src="https://cdn.pixabay.com/photo/2016/11/22/23/13/black-dog-1851106__340.jpg"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
